web/src/people.js

```diff
@@ -58,9 +59,17 @@ export function get_users_from_ids(user_ids) {
     return user_ids.map((user_id) => get_by_user_id(user_id));
 }
 
-export function get_by_user_id(user_id, ignore_missing) {
+// Use this function only when you are sure that user_id is valid.
+export function get_by_user_id(user_id) {
+    const person = people_by_user_id_dict.get(user_id);
+    assert(person, "Unknown user_id in get_by_user_id: " + user_id);
+    return person;
+}
+
+// This is type unsafe version of get_by_user_id for the callers that expects undefined values.
+export function maybe_get_user_by_id(user_id, ignore_missing) {
```

Comment:
I think `ignore_missing` is no longer needed. This function will just `return people_by_user_id_dict.get(user_id)`, and we can call it in the new `get_by_user_id` with the assertion.

Reply:
@PIG208 Then what about the blueslip call we have in `get_by_user_id` ?
For example(compose_validate.js) calls like this-
```javascript
export function needs_subscribe_warning(user_id, stream_id) {
    // This returns true if all of these conditions are met:
    //  * the user is valid
    //  * the user is not already subscribed to the stream
    //  * the user has no back-door way to see stream messages
    //    (i.e. bots on public/private streams)
    //
    //  You can think of this as roughly answering "is there an
    //  actionable way to subscribe the user and do they actually
    //  need it?".
    //
    //  We expect the caller to already have verified that we're
    //  sending to a valid stream and trying to mention the user.

    const user = people.get_by_user_id(user_id);

    if (!user) {
        return false;
    }
```
Handles undefined values but it should also report the missing user id via blueslip call right ?

Reply:
And then there are these calls for example in `rendered_markdown.js` line 93 which intentionally wants to skip blueslip errors but acknowledges the undefined values.
```javascript
const person = people.get_by_user_id(user_id, true);
if (person !== undefined) {
    // Note that person might be undefined in some
    // unpleasant corner cases involving data import.
    set_name_in_mention_element(this, person.full_name);
}
```
And hence we had the `ignore_missing` parameter for such cases.

Reply:
For those cases where `ignore_missing` is set to true, we can use `maybe_get_by_user_id` to preserve the behavior. As for the blueslip call, I agree that the check should be kept. So let's just remove `ignore_missing` and always log the error when the user id is not there.

Reply:
Perhaps in the future, with more modules migrated to TypeScript, it will be easier for us to determine in which cases there might be missing user ids.